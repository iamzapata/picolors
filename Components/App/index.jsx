import React, { useState } from "react"

import Header from "Components/Header"
import Canvas from "Components/Canvas"
import PiDigitCount from "Components/PiDigitCount"
import ColorSelector from "Components/ColorSelector"
import PaletteSelector from "Components/PaletteSelector"

import GridStyles from "styles/grids/Grid_10_Items.module.scss"

import { regular, grey, brown, pastel } from "utils/colorMap"

const App = () => {
  const colorPalettes = [regular, grey, brown, pastel]
  const [currentPalette, setCurrentPalette] = useState(regular)
  const originalPaletteColors = colorPalettes.find(
    (palette) => palette.type === currentPalette.type
  ).colors
  const selectedPaletteColors = currentPalette.colors

  const handleColorChange = (index, color) => {
    setCurrentPalette({
      ...currentPalette,
      colors: {
        ...currentPalette.colors,
        [index]: color,
      },
    })
  }

  const handlePaletteSelectChange = (event) => {
    const { value } = event.target
    setCurrentPalette(colorPalettes.find((palette) => palette.type === value))
  }

  return (
    <div className="App d-flex">
      <PiDigitCount />

      <div>
        <Header />
        <Canvas width={800} height={800} colorMap={selectedPaletteColors} />
      </div>

      <div className="m-3">
        <PaletteSelector
          colorPalettes={colorPalettes}
          handlePaletteSelectChange={handlePaletteSelectChange}
        />

        <div className={GridStyles.Grid_3_3}>
          {Object.values(currentPalette.colors).map((rgb, index) => (
            <ColorSelector
              key={index}
              rgb={rgb}
              index={index}
              originalColor={originalPaletteColors[index]}
              handleChange={handleColorChange}
            />
          ))}
        </div>
      </div>
    </div>
  )
}

export default App
