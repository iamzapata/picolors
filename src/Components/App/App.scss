html,
body,
#app,
.App {
  height: 100%;
}

body {
  margin: 0;
  padding: 0;
  width: 100%;
  display: table;
  font-weight: 100;
  font-family: "Lato";
}

.App {
  display: flex;
  align-items: center;
  justify-content: center;
}

button {
  font-family: "Iceland", cursive !important;
}
