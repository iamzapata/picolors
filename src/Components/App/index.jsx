import React, { Component } from "react"

import Canvas from "Components/Canvas"
import ColorSelector from "Components/ColorSelector"
import "./App.scss"

import { pastel, regular } from "utils/colorMap"

class App extends Component {
  state = {
    colorPalettes: [regular, pastel],
    currentPalette: regular
  }

  handleColorChange = (index, color) => {
    const { colorPalettes, currentPalette } = this.state

    const newPalettes = colorPalettes.map(palette => {
      if (palette.type === currentPalette.type) {
        return {
          ...palette,
          colors: {
            ...palette.colors,
            [index]: color
          }
        }
      }
      return palette
    })

    this.setState({
      colorPalettes: newPalettes
    })
  }

  handlePaletteSelectChange = palette => {
    return () => {
      this.setState({ currentPalette: palette })
    }
  }

  render() {
    const { colorPalettes, currentPalette } = this.state

    const selectedPaletteColors = colorPalettes.find(
      palette => palette.type === currentPalette.type
    ).colors

    return (
      <div className="App">
        <Canvas width={1000} height={1000} colorMap={selectedPaletteColors} />
        <div className="d-flex w-25" style={{ justifyContent: "space-evenly" }}>
          {colorPalettes.map(palette => (
            <div
              key={palette.type}
              className="d-flex flex-column mr-3 align-items-center justify-content-center"
            >
              <h4 className="">{palette.type}</h4>
              <input
                style={{ height: "30px", width: "30px" }}
                className="form-control ml-2"
                type="radio"
                checked={currentPalette.type === palette.type}
                onChange={this.handlePaletteSelectChange(palette)}
              />
              {Object.values(palette.colors).map((rgb, index) => (
                <ColorSelector
                  rgb={rgb}
                  key={index}
                  index={index}
                  handleChange={this.handleColorChange}
                  enabled={currentPalette.type === palette.type}
                />
              ))}
            </div>
          ))}
        </div>
      </div>
    )
  }
}

export default App
