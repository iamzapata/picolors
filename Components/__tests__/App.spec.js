import { render } from '@testing-library/react'
import App from 'Components/App'

jest.mock('Components/Header', () => () => <div>Header</div>)
jest.mock('Components/Header', () => () => <div>Canvas</div>)
jest.mock('Components/PiDigitCount', () => () => <div>PiDigitCount</div>)
jest.mock('Components/ColorSelector', () => () => <div>ColorSelector</div>)
jest.mock('Components/PaletteSelector', () => () => <div>PaletteSelector</div>)

describe('<App />', () => {
  it('renders', () => {
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })
})
