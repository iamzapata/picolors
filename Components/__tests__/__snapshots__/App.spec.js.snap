// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
<div>
  <div
    class="App d-flex"
  >
    <div>
      PiDigitCount
    </div>
    <div>
      <div>
        Canvas
      </div>
      <canvas
        height="1000"
        width="1000"
      />
    </div>
    <div
      class="m-3"
    >
      <div>
        PaletteSelector
      </div>
      <div>
        <div>
          ColorSelector
        </div>
        <div>
          ColorSelector
        </div>
        <div>
          ColorSelector
        </div>
        <div>
          ColorSelector
        </div>
        <div>
          ColorSelector
        </div>
        <div>
          ColorSelector
        </div>
        <div>
          ColorSelector
        </div>
        <div>
          ColorSelector
        </div>
        <div>
          ColorSelector
        </div>
        <div>
          ColorSelector
        </div>
      </div>
    </div>
  </div>
</div>
`;
